internal/ecrconsumer/image_values_test.go

```diff
@@ -302,12 +302,31 @@ func TestWithImage(t *testing.T) {
 		t.Fatal("no matching image found")
 	}
 
-	imgVals := table(outputRls.Spec.Values, path)
+	outputImg := outputRls.Spec.Values["image"].(map[string]interface{})
+	assert.Equal(t, expectedImg.Tag, outputImg["tag"].(string))
 
-	outputImage, err := parseImage(imgVals["repository"].(string), imgVals["tag"].(string))
-	assert.NoError(t, err)
+	// Test No Matching Image Case
+	rls = v1alpha1.HelmRelease{
```

Comment:
The match/no match cases should be done in separate sub-tests, so it's obvious what actually caused `TestWithImage` to fail. Usually I'd split them into separate test funcs, but since they share some test data I think it makes sense to group them together.

We can create a sub-test scope with 
```
t.Run("descriptive name", func(t *testing.T) { 
    // sub test
})
```